JS SDK Revamp (#725)
rudderlabs/rudderstack-docs: docs/sources/sdks/rudderstack-javascript-sdk/index.mdx

---
title: "JavaScript"
description: >-

>    Detailed technical documentation on RudderStack JavaScript SDK, its implementatation and usage.

Reviewer: ```suggestion
Detailed technical documentation on the RudderStack JavaScript SDK, its implementation, and usage.
```

aliases: ["JavaScript", "JavaScript SDK", "JavaScript source", "JS SDK", "JS source", "Web SDK", "web source"]
---

RudderStack's JavaScript SDK is the most popular and widely used SDK to track the customer event data from your website and send it to the specified destinations.

<GhBadge
  url={'https://www.npmjs.com/package/rudder-sdk-js'}
  repo={'npm/v/rudder-sdk-js'}
/>

<div class="infoBlock">
Find the open source code for the <a href="https://github.com/rudderlabs/rudder-sdk-js">JavaScript SDK</a> on GitHub.
</div>

> Refer to the following guides to know about the usage and features of JavaScript SDK:

Reviewer: ```suggestion
Refer to the following guides to know more about the usage and features of the JavaScript SDK:
```

| Guide          | Description    |
| :------------------- | :------------------------- |

> | <Link to="/sources/sdks/rudderstack-javascript-sdk/quick-start-guide/">Quickstart</Link>    | Get started on how to set up and start using the JavaScript SDK. |

Reviewer: ```suggestion
| <Link to="/sources/sdks/rudderstack-javascript-sdk/quick-start-guide/">Quickstart</Link>    | Get started on setting up and using the JavaScript SDK. |
```

> | <Link to="/sources/sdks/rudderstack-javascript-sdk/load-js-sdk/">Load SDK</Link>    | Detailed steps to load the JavaScript SDK using different options. |

Reviewer: ```suggestion
| <Link to="/sources/sdks/rudderstack-javascript-sdk/load-js-sdk/">Load SDK</Link>    | Detailed steps on loading the JavaScript SDK using different options. |
```

| <Link to="/sources/sdks/rudderstack-javascript-sdk/supported-api/">Supported API</Link>                     | Supported API calls to track the event data.      |

> | <Link to="/sources/sdks/rudderstack-javascript-sdk/version-migration-guide/">Version Migration</Link>    | Detailed steps to migrate the SDK from v1 to v1.1.     |

Reviewer: ```suggestion
| <Link to="/sources/sdks/rudderstack-javascript-sdk/version-migration-guide/">Version Migration Guide</Link>    | Detailed steps on migrating the SDK from v1 to v1.1.     |
```

> |  <Link to="/sources/sdks/rudderstack-javascript-sdk/data-storage-cookies/">Data Storage</Link>                       | Persistent user data storage in cookies and local storage.                 |

Reviewer: ```suggestion
|  <Link to="/sources/sdks/rudderstack-javascript-sdk/data-storage-cookies/">Data Storage</Link>                       | Details on the persistent user data stored in cookies and local storage.                 |
```

> |  <Link to="/sources/sdks/rudderstack-javascript-sdk/filtering/">Filtering</Link>                      | Methods to filter events and destinations while sending events.   |

Reviewer: ```suggestion
|  <Link to="/sources/sdks/rudderstack-javascript-sdk/filtering/">Event Filtering</Link>                      | Methods to filter events and destinations while sending events.   |
```

> |  <Link to="/sources/sdks/rudderstack-javascript-sdk/onetrust-consent-manager/">OneTrust Consent Manager</Link>     |  Integrate JavaScript SDK with OneTrust consent management platform.      |

Reviewer: ```suggestion
|  <Link to="/sources/sdks/rudderstack-javascript-sdk/onetrust-consent-manager/">OneTrust Consent Manager</Link>     |  Detailed steps on integrating the JavaScript SDK with the OneTrust consent management platform.      |
```

|   <Link to="/sources/sdks/rudderstack-javascript-sdk/detecting-adblocked-pages/">Detecting Ad-blocked Pages</Link>                     | Detect ad-blocked pages using the JavaScript SDK.                    |

> | <Link to="/sources/sdks/rudderstack-javascript-sdk/javascript-sdk-enhancements/">Enhancements</Link>       | Use Beacon utility to send event payloads using the JavaScript SDK.            |

Reviewer: ```suggestion
| <Link to="/sources/sdks/rudderstack-javascript-sdk/javascript-sdk-enhancements/">Enhancements</Link>       | Detailed guidance on using the Beacon utility to send event payloads using the JavaScript SDK.      |
```

> | <Link to="/sources/sdks/rudderstack-javascript-sdk/faq/">FAQ</Link>       | Commonly asked questions and their solutions. |

Reviewer: ```suggestion
| <Link to="/sources/sdks/rudderstack-javascript-sdk/faq/">FAQ</Link>       | Commonly asked questions on the JavaScript SDK and their solutions. |
```

> | Chromecast     | RudderStack supports integrating the JavaScript SDK with [**Google Chromecast**](https://store.google.com/in/product/chromecast?hl=en-GB). You can build the web sender app by following <a href="https://developers.google.com/cast/docs/web_sender">these instructions</a> and then adding the JavaScript SDK to it. Refer to the <a href="https://developers.google.com/cast/docs/developers">Google Cast developer guide</a> for more details.|

Reviewer: ```suggestion

<div class="successBlock">
RudderStack also supports integrating the JavaScript SDK with <a href="https://store.google.com/in/product/chromecast?hl=en-GB">Google Chromecast</a>. You can build the web sender app by following <a href="https://developers.google.com/cast/docs/web_sender">these instructions</a> and adding the JavaScript SDK. Refer to the <a href="https://developers.google.com/cast/docs/developers">Google Cast developer guide</a> for more details.
</div>
```

Reviewer: As it's a side note, let's just have it as a success block instead of making it a part of the table. WDYT?
